datalog.go

```diff
@@ -79,15 +79,15 @@ func parseSegmentName(name string) (uint16, uint64, error) {
 }
 
 func (dl *datalog) openSegment(name string, id uint16, seqID uint64) (*segment, error) {
-	f, err := openFile(dl.opts.FileSystem, name, false)
+	f, err := openFile(dl.opts.FileSystem, name, false, dl.opts.ReadOnly)
```

Comment:
Wondering if rather than modifying all `openFile` calls it would be cleaner to implement a new file system wrapper similar to [fs.Sub](https://github.com/akrylysov/pogreb/blob/master/fs/sub.go#LL8C23-L8C23)? The wrapper would set `os.O_RDONLY` flags and implement a noop `CreateLockFile`.